File: Excel/300-399/314/314 Challenge.py
import pandas as pd

path = "300-399/313/313 Atbash Palindromes.xlsx"
input_df = pd.read_excel(path, usecols="A", nrows=9)
test = pd.read_excel(path, usecols="B", nrows=3)


def is_atbash_palindrome(word):
    vals = [ord(c) - ord("A") + 1 for c in word]
    return all(a + b == 27 for a, b in zip(vals, reversed(vals)))


result = (
    input_df[input_df["Text"].apply(is_atbash_palindrome)][["Text"]]
    .reset_index(drop=True)
)

print(result["Text"].equals(test["Answer Expected"]))
# True

File: Excel/300-399/314/314 Challenge.R
library(tidyverse)
library(readxl)

path <- "300-399/313/313 Atbash Palindromes.xlsx"
input <- read_excel(path, range = "A1:A10")
test  <- read_excel(path, range = "B1:B4")

is_atbash_palindrome <- function(word) {
  chars <- utf8ToInt(word) - utf8ToInt("A") + 1L
  all(chars + rev(chars) == 27L)
}

result <- input %>%
  filter(map_lgl(Text, is_atbash_palindrome)) %>%
  select(Text)

all.equal(result$Text, test$`Answer Expected`)
# [1] TRUE
